src/bin/lpc55/main.rs

```diff
@@ -41,11 +41,17 @@ fn try_main(args: clap::ArgMatches<'_>) -> anyhow::Result<()> {
         2 => log::set_max_level(log::LevelFilter::Debug),
         _ => log::set_max_level(log::LevelFilter::Trace),
     };
+    
+    let pid = match args.value_of("PID") {
+        Some(pid) => Some(u16::from_str_radix(pid.trim_start_matches("0x"), 16).expect("Could not parse PID")),
+        _ => None
```

Comment:
this should be:
```
let pid = arg.value_of("PID")
    .map(|pid| u16::from_str_radix(pid.trim_start_matches("0x"), 16))
    .map_err(|e| anyhow!("Could not parse PID"))?;
```
same for vid.

Reply:
I tried getting this to work and this is the closest I came up with.

```rust
    let pid = args.value_of("PID").map( |pid|
        u16::from_str_radix(pid.trim_start_matches("0x"), 16).map_err(|_| anyhow!("Could not parse PID"))?
    );
```

It doesn't work since the error cannot convert to a None type error (since the resulting type is an `Option<T>`, whereas it works in other places that would return a `Result`.  So I am keeping the match statements, but at least i added the `anyhow!`.

Reply:
You can convert `Result<T, E>` into `Option<T>` with `.ok()`. And the other way with `.ok_or<E>(self, err: E)`.